src/AuthenticationPortal.AwsExtension/AwsCognito.cs

```diff
@@ -40,7 +45,18 @@ public async Task<SignInResponse> SignInAsync(SignInRequest signInRequest)
                     AccessToken = authResponse.AuthenticationResult.AccessToken
                 };
                 GetUserResponse getUser = await providerClient.GetUserAsync(getUserRequest);
-                _signInResponse.UserId = getUser.UserAttributes[0].Value;
+                Contracts.AddUserRequest addUserRequest = new Contracts.AddUserRequest()
+                {
+                    UserId = getUser.UserAttributes[1].Value,
+                    FirstName = getUser.UserAttributes.Where(a => a.Name == "custom:Firstname").First().Value,
+                    LastName = getUser.UserAttributes.Where(a => a.Name == "custom:Lastname").First().Value,
+                    ContactNumber = getUser.UserAttributes.Where(a => a.Name == "phone_number").First().Value,
+                    Email = getUser.UserAttributes.Where(a => a.Name == "email").First().Value,
+
+                };
+
+                var response = await UserDetails.SaveAsync(addUserRequest, _userStore);
```

Comment:
this should be handled in service layer, awscongnito's responsibility is to sign in the user and return the details